src/master-playlist-controller.js

```diff
@@ -354,18 +354,18 @@ export class MasterPlaylistController extends videojs.EventTarget {
       new SegmentLoader(videojs.mergeOptions(segmentLoaderOptions, {
         segmentMetadataTrack: this.segmentMetadataTrack_,
         loaderType: 'main'
-      }));
+      }), options);
```

Comment:
Because that are distinctly different "options" one comes from outside the project and the other is more of an object with functions necessary to the correct functioning of SegmentLoader.

Merging them feels very wrong.